docs/pull-request-template.md

```diff
@@ -0,0 +1,49 @@
+# Pull request template
```

Comment:
Because the template links to this doc, it should be included either in the same commit or a preceding one.

I think making it the preceding commit is probably ideal -- this doc is about making a good PR, and doesn't really need to rely on the template.